main/parsers/zettle_parser.py

```diff
@@ -1,71 +1,89 @@
 # Use to generate the JSON-object from the Zettle API respnose.
-import sys, os
-
-from datetime import datetime, timedelta, time
-import pytz
-from dateutil.tz import tzlocal
-
-from requests_oauthlib import OAuth2Session
 import json
+import os
+import time
+from datetime import datetime, timedelta
+
 import dateutil.parser
+import requests
+from dateutil.tz import tzlocal
 from flatten_dict import unflatten
-
-from parsers.parser import Parser
 from helpers.asset_loader import AssetLoader
+from parsers.parser import Parser
 
 al = AssetLoader()
 
 
 class ZettleParser:  # Parser
     def __init__(self):
-        self.zettle = self.set_cred()
+        self.auth_token = None
 
     def intakt_type(self):
         return "Zettle"
 
-    def set_cred(self):
-        access_file = (
+    def get_new_auth_token(self):
+        access_file_path = (
             os.path.dirname(os.path.realpath(__file__)) + "/../credentials/access.json"
         )
+        with open(access_file_path) as access_file:
+            access_json = json.load(access_file)
 
-        with open(access_file) as f:
-            access_cred = json.load(f)
-            f.close()
+        token_response = requests.post(
+            "https://oauth.zettle.com/token",
+            headers={"Content-Type": "application/x-www-form-urlencoded"},
+            data=f"grant_type={access_json['grant_type']}&client_id={access_json['client_id']}&assertion={access_json['assertion']}",
+        )
+
+        if token_response.ok:
+            response_json = token_response.json()
+            self.auth_token = response_json["access_token"]
+            response_json["expires_at"] = time.time() + response_json["expires_in"]
 
-        client_id = access_cred["client_id"]
-        redirect_uri = "https://httpbin.org/get"
+            saved_token_file_path = (
+                os.path.dirname(os.path.realpath(__file__))
+                + "/../credentials/last_used_zettle_token.json"
+            )
+            with open(saved_token_file_path, "w") as saved_token_file:
+                json.dump(response_json, saved_token_file)
```

Comment:
`saved_token_file` should be closed here.

Reply:
See comment above